### PR TITLE
Added IPv6 support for radius_server

### PR DESCRIPTION
**Pull request by jonnytdevops**
Folks, it would be appreciated if you could review this for IPv6 support, thanks

### CODE REVIEW

**Comment by jonnytdevops** [comment]
TestRadiusServer#test_create_destroy_single_ipv6 = 14.73 s = .
TestRadiusServer#test_create_destroy_single = 14.00 s = .
TestRadiusServer#test_radiusserver_create_destroy_multiple = 16.23 s = .

Finished in 44.970864s, 0.0667 runs/s, 1.8901 assertions/s.

3 runs, 85 assertions, 0 failures, 0 errors, 0 skips

**Comment by jonnytdevops** [comment]
Note: the reason why we have 2 validation steps for the IP address, is because we don't want to allow subnets, which unfortunately ruby's ipaddr library accepts and returns true for the ipv4? and ipv6? methods.

File: lib/cisco_node_utils/radius_server.rb [hunk 1]
```diff
@@ -25,8 +26,16 @@ class RadiusServer < NodeUtil
     attr_reader :name
 
     def initialize(name, instantiate=true)
-      unless name[/^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$/]
-        fail ArgumentError, 'Invalid value (Name is not an IP address)'
+      unless name =~ /^[a-zA-Z0-9\.\:]*$/
+        fail ArgumentError,
+             'Invalid value (IP address contains invalid characters)'
```
Comment (chrisvanheuveln) [inline lib/cisco_node_utils/radius_server.rb]: `~IP~IPv4/IPv6~`

Line 38 as well.

File: tests/test_radius_server.rb [hunk 2]
```diff
@@ -90,9 +91,60 @@ def test_radiusserver_create_destroy_single
     refute_includes(Cisco::RadiusServer.radiusservers, id)
   end
 
+  def test_create_destroy_single_ipv6
+    id = '2004::3'
+    refute_includes(Cisco::RadiusServer.radiusservers, id)
+
+    server = Cisco::RadiusServer.new(id, true)
+    assert_includes(Cisco::RadiusServer.radiusservers, id)
+    assert_equal(Cisco::RadiusServer.radiusservers[id], server)
```
Comment (chrisvanheuveln) [inline tests/test_radius_server.rb]: You've got the params reversed on these `assert_equal` calls - should be `assert_equal(expected, actual)`

When they fail they'll display the wrong values for `expected:` and `actual:`

**Comment by jonnytdevops** [comment]
@chrisvanheuveln Code updated. Can this be merged now please? Thanks

**Comment by jonnytdevops** [comment]
TestRadiusServer#test_create_destroy_single = 14.58 s = .
TestRadiusServer#test_create_destroy_single_ipv6 = 13.83 s = .
TestRadiusServer#test_radiusserver_create_destroy_multiple = 15.82 s = .

Finished in 44.226840s, 0.0678 runs/s, 1.9219 assertions/s.

3 runs, 85 assertions, 0 failures, 0 errors, 0 skips

**Comment by jonnytdevops** [comment]
@chrisvanheuveln just realised that I've swapped everything when it didn't need to be swapped. Just re-looking over this, thanks.

**Comment by jonnytdevops** [comment]
@chrisvanheuveln actually this looks good to me now! Please merge when ready, unless you see any other issues :)

BTW, assert_includes was always correct. The function fails the other way round :)

Chris: yep, I was only concerned about the assert_equal checks :-)

**Pull request merged**